github-pullrequest-plugin/src/main/java/org/jenkinsci/plugins/github/pullrequest/trigger/JobRunnerForCause.java

```diff
@@ -115,23 +120,40 @@ public boolean apply(GitHubPRCause cause) {
     /**
      * Cancel previous builds for specified PR id.
      */
-    private static boolean cancelQueuedBuildByPrNumber(final int id) {
+    protected int cancelQueuedBuildByPrNumber(final int id) {
         Queue queue = getJenkinsInstance().getQueue();
+        //todo replace with stream?
+        int canceled = 0;
+        for (Queue.Item item : queue.getItems()) {
+            if (!(item.task instanceof Job)) {
+                LOGGER.debug("Item {} not instanceof job", item);
+                continue;
+            }
+
+            final Job<?, ?> jobTask = (Job<?, ?>) item.task;
+            if (!jobTask.getDisplayName().equals(job.getDisplayName())) {
```

Comment:
Do not do display name comparisons. You may compare `fullName`s, or simply use pointer equality to compare `Item`s.